### PR TITLE
rejudge problem

### PR DESCRIPTION
**Pull request by geniusgordon**
rejudge problem

### CODE REVIEW

File: problem/static/problem/js/panel.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+$(document).ready(function() {
+    $('[data-toggle="tooltip"]').tooltip();
+    $('.progress').css('overflow', 'inherit');
+    $('.del_btn').click(function() {
+        return confirm("Are you sure you want to delete?");
+    });
+    $('.rejudge-btn').click(function() {
```
Comment (nickelcheng) [inline problem/static/problem/js/panel.js]: can you use identical naming rule? ```.del_btn``` or ```.rejudge-btn```

File: problem/views.py [hunk 2]
```diff
@@ -290,3 +291,20 @@ def download_special(request, filename):
     response['Content-Disposition'] = 'attachment; filename=' + filename
     return response
 
+def rejudge(request):
```
Comment (nickelcheng) [inline problem/views.py]: ```@login_required()``` and auth checking is needed.

File: problem/views.py [hunk 3]
```diff
@@ -290,3 +291,20 @@ def download_special(request, filename):
...
+            problem = Problem.objects.get(pk=pid)
+            if not has_problem_ownership(request.user, problem) and \
+                    not request.user.has_admin_auth():
+                logger.warning("%s has no permission to rejudge problem %d"
+                        % (request.user, problem.pk))
+                raise PermissionDenied()
+            rejudge_problem(problem)
+            logger.info("problem %s rejudged" % problem.pk)
+        except Problem.DoesNotExist:
+            print pid
```
Comment (nickelcheng) [inline problem/views.py]: Please remove this.

File: problem/static/problem/js/panel.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        var submission = $(this).attr('data-submission');
+        return confirm("\
```
Comment (nickelcheng) [inline problem/static/problem/js/panel.js]: I suggest to force user to enter number of submissions.
Rejudge the problem only if the user enter correct number.

**Comment by geniusgordon** [comment]
fixed

File: problem/views.py (hunk 2 above)
Comment (nickelcheng) [inline problem/views.py]: auth checking is needed, only user who has problem auth can rejudge problem

**Comment by nickelcheng** [comment]
still 2 comments are not fixed

File: problem/static/problem/js/panel.js (hunk 1 above)
Comment (geniusgordon) [inline problem/static/problem/js/panel.js]: fixed

**Comment by geniusgordon** [comment]
fixed

**Comment by nickelcheng** [comment]
No matter rejudge successful or not, please show a message.
Use @Changron's message in #132

**Comment by geniusgordon** [comment]
add message

**Comment by CHIAHSINHOU** [comment]
This PR is ok for me.

File: problem/static/problem/js/panel.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+$(document).ready(function() {
```
Comment (nickelcheng) [inline problem/static/problem/js/panel.js]: Please add license

File: problem/static/problem/js/panel.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    $('.del-btn').click(function() {
+        return confirm("Are you sure you want to delete?");
+    });
+    $('.rejudge-btn').click(function() {
+        var submission = $(this).attr('data-submission');
+        var check = prompt("\
+		The problem has " + submission + " submissions\n\
+		Rejudge might take a long time\n\
+		Enter the submission number of this problem to continue");
+	return check == submission;
```
Comment (nickelcheng) [inline problem/static/problem/js/panel.js]: if submission not match, give another alert says that rejudge doesn't start

**Comment by nickelcheng** [comment]
+1 besides the comment.
@bruce3557 please check

File: problem/static/problem/js/panel.js (hunk 5 above)
Comment (geniusgordon) [inline problem/static/problem/js/panel.js]: added

File: problem/static/problem/js/panel.js (hunk 6 above)
Comment (geniusgordon) [inline problem/static/problem/js/panel.js]: added

**Comment by bruce3557** [comment]
+1

**Pull request merged**